gh-106505: don't lookup own hostname in getfqdn()

bluelhf opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

- return `gethostname()` immediately instead of resolving it

<!-- gh-issue-number: gh-106505 -->
* Issue: gh-106505
<!-- /gh-issue-number -->

bluelhf committed:
fix(socket): don't lookup own hostname in getfqdn()

bluelhf committed:
chore(blurb): add news entry

bluelhf committed:
chore(acks): add self

serhiy-storchaka reviewed (commented):
Does it covered by existing tests? If no, please add tests.

Also, please correct the NEWS entry. The sentence should start with a title character and end with a period.

bluelhf committed:
fix(blurb): reword news entry

NEWS entries must start with a title case letter and end with a period — the entry was also rewritten to match the usual wording of entries
